### ISSUE TITLE
Crash on Ctrl+Tab for DockSpaceOverViewport with PassthruCentralNode flag

### ISSUE THREAD

**Issue by ilovexyz** [ocornut/imgui#4792]
**Version/Branch of Dear ImGui:**

Version: 1.86 WIP (18516)
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_glfw.cpp + imgui_impl_opengl3.cpp 
Compiler: MSVC
Operating System: Windows 10

**My Issue/Question:**

I want to use the ```ImGui::DockSpaceOverViewport``` function in my application, but it will cause crash when pressing Ctrl+Tab navigation key, i wonder if this is a bug or i didn't understand the usage of dockspace.

**Screenshots/Video**

Step to reproduce the issue based on the ```example_glfw_opengl3```:

<img width="1283" alt="exception" src="https://user-images.githubusercontent.com/18619226/145219034-0ee3b74f-9aee-4157-9cec-a853b7443f5d.gif">

Code crashes here:

<img width="824" alt="exception" src="https://user-images.githubusercontent.com/18619226/145219028-6597c873-592b-49c6-a7e7-901e204c1aa7.png">

**Standalone, minimal, complete and verifiable example:** 

Just add the following code to  the main.cpp of ```example_glfw_opengl3``` will reproduce the issue.
```
// show dock
ImGui::DockSpaceOverViewport(ImGui::GetMainViewport(), ImGuiDockNodeFlags_PassthruCentralNode);
```
<img width="799" alt="code" src="https://user-images.githubusercontent.com/18619226/145219024-146ef94b-d11e-4cb2-9ec7-e210b4549436.png">

**Comment by ocornut** [comment]
This is code we changed recently. Can you confirm if updating to very latest commit still reproduce the issue?

**Comment by ilovexyz** [comment]
> 

yes, i just pull the lastest docking branch, the above issue is running using the lastest code

**Comment by ilovexyz** [comment]
> This is code we changed recently. Can you confirm if updating to very latest commit still reproduce the issue?

if add the following code before accessing ```CmdBuffer.Data```, it will crash elsewhere.
```
    if (CmdBuffer.Size == 0)
        return;
```

from the debug stack, i can see the value of ```CmdBuffer.Size``` is zero, but i don't know how to correct this.

**Comment by ocornut** [comment]
Should be fixed by 0647cf4, thank you very much for reporting and helping!

**Comment by ilovexyz** [comment]
wow, thank you for the quick response, i just pulled and checked, no crash any more!

**Comment by RT222** [comment]
I stumbled upon the same issue while migrating to imgui 1.87 wip. It seems to happens only with empty windows. You can reproduce it like this :
```
const ImGuiID dock_id = ImGui::DockSpaceOverViewport(ImGui::GetMainViewport());
ImGui::SetNextWindowDockID(dock_id, ImGuiCond_FirstUseEver);
ImGui::Begin("Window");
ImGui::End();
```

A Begin followed by and End call will always result in a crash if the window is docked and reached with ctrl+tab. This can also be reproduced with the log or long text display windows of the imgui demos because theses windows have such a code path.

**Comment by ocornut** [comment]
Thank you, will look into fixing soon.

**Comment by ocornut** [comment]
@RT222 I cannot reproduce a crash. Can you post details and a callstack?

I however noticed that during CTRL+Tabbing the background color of the viewport will start appearing above non-docked windows which is most likely due to the BgColor/dimming changes done in 1.86.

**Comment by RT222** [comment]
Using the latest docking commit (https://github.com/ocornut/imgui/commit/08f3aa89723f8e41a8b10aa0c60c6617da563e18) and the default imgui examples files without any modification, I was able to reproduce this crash consistently on all win32 and sdl2 based backends by docking the log window on the main viewport.

Then I tried to test more build configurations and I think I found why you couldn't reproduce it on your end : it only happens on 64 bits builds. Try it with any backend built on debug/64bits, I'm pretty confident you should get this crash.

Here's my build infos : 
```
Dear ImGui 1.87 WIP (18612)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=199711
define: _WIN32
define: _WIN64
define: _MSC_VER=1930
define: _MSVC_LANG=201402
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: imgui_impl_win32
io.BackendRendererName: imgui_impl_dx11
io.ConfigFlags: 0x00000441
 NavEnableKeyboard
 DockingEnable
 ViewportsEnable
io.ConfigViewportsNoDecoration
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x00001C0E
 HasMouseCursors
 HasSetMousePos
 PlatformHasViewports
 HasMouseHoveredViewport
 RendererHasVtxOffset
 RendererHasViewports
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 1264.00,761.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

**Comment by ocornut** [comment]
Thank you I have managed to repro. It is possible to get a better assert with `#define IMGUI_DEBUG_PARANOID` in imconfig.h

Pushed a fix f1a0731
It's embarrassing how many regressions those window dimming/bgcolor system changes caused :(

Leaving this open because of :
> I however noticed that during CTRL+Tabbing the background color of the viewport will start appearing above non-docked windows which is most likely due to the BgColor/dimming changes done in 1.86.

**Comment by ocornut** [comment]
> Leaving this open because of :
> I however noticed that during CTRL+Tabbing the background color of the viewport will start appearing above non-docked windows which is most likely due to the BgColor/dimming changes done in 1.86.

Moving this to #5917 as main issue here has been fixed a while ago.
